components/style/gecko/wrapper.rs

```diff
@@ -1574,30 +1592,18 @@ impl<'le> ::selectors::Element for GeckoElement<'le> {
         self.get_state().intersects(NonTSPseudoClass::AnyLink.state_flag())
     }
 
-    fn get_id(&self) -> Option<Atom> {
-        if !self.has_id() {
-            return None;
-        }
-
-        let ptr = unsafe {
-            bindings::Gecko_AtomAttrValue(self.0,
-                                          atom!("id").as_ptr())
-        };
-
-        if ptr.is_null() {
-            None
-        } else {
-            Some(Atom::from(ptr))
-        }
+    fn has_id(&self, id: &Atom, case_sensitivity: CaseSensitivity) -> bool {
+        self.get_id().map_or(false, |atom| case_sensitivity.eq_atom(&atom, id))
```

Comment:
We could tweak this to skip the refcount traffic from get_id(), which would probably be a significant win given that the atom refcounting is atomic. Can you either fix that here or file a followup?